CSharpMath.Editor.Tests/CaretTests.cs

```diff
@@ -176,4 +177,107 @@ public class CaretCanStartAndStopBlinking {
+      Assert.Equal(System.Drawing.Color.Blue, inner.Color);
+
+      await Task.Delay((int)MathKeyboard<TestFont, char>.DefaultBlinkMilliseconds + CaretBlinks.MillisecondBuffer);
+      Assert.Equal(MathKeyboardCaretState.ShownThroughPlaceholder, keyboard.CaretState);
+      Assert.Equal("😐", outer.Nucleus);
+      Assert.Equal(System.Drawing.Color.Blue, outer.Color);
+      Assert.Equal("😀", inner.Nucleus);
+      Assert.Equal(System.Drawing.Color.Green, inner.Color);
+
+      RestorePlaceholderDefaultSettings();
+    }
+    [Fact]
+    public void AllCustomizablePlaceholderPropertiesAreResetOnCaretVisible() {
+      LaTeXSettings.PlaceholderActiveNucleus = "😀";
+      LaTeXSettings.PlaceholderRestingNucleus = "😐";
+      LaTeXSettings.PlaceholderActiveColor = System.Drawing.Color.Green;
+      LaTeXSettings.PlaceholderRestingColor = System.Drawing.Color.Blue;
+      var keyboard = new MathKeyboard<TestFont, char>(TestTypesettingContexts.Instance, new TestFont()) {
+        CaretState = MathKeyboardCaretState.Shown
+      };
+      Assert.Equal(MathKeyboardCaretState.Shown, keyboard.CaretState);
+      keyboard.KeyPress(MathKeyboardInput.Subscript);
+      var outer = Assert.IsType<Atom.Atoms.Placeholder>(Assert.Single(keyboard.MathList));
+      var inner = Assert.IsType<Atom.Atoms.Placeholder>(Assert.Single(outer.Subscript));
+      Assert.Equal(MathKeyboardCaretState.ShownThroughPlaceholder, keyboard.CaretState);
+
+      keyboard.InsertionIndex = MathListIndex.Level0Index(keyboard.MathList.Count);
+      Assert.Equal(MathKeyboardCaretState.Shown, keyboard.CaretState);
+      Assert.Equal(LaTeXSettings.PlaceholderRestingNucleus, outer.Nucleus);
+      Assert.Equal(LaTeXSettings.PlaceholderRestingColor, outer.Color);
+      Assert.Equal(LaTeXSettings.PlaceholderRestingNucleus, inner.Nucleus);
+      Assert.Equal(LaTeXSettings.PlaceholderRestingColor, inner.Color);
+      RestorePlaceholderDefaultSettings();
+    }
+    [Fact]
+    public void LaTeXSettings_Placeholder_IsNewInstance() {
+      Assert.False(LaTeXSettings.Placeholder == LaTeXSettings.Placeholder);
+      // Double check, also verify that its contents are 'fresh':
+      LaTeXSettings.Placeholder.Nucleus = "x";
+      Assert.Equal(RestingNucleusDefault, LaTeXSettings.Placeholder.Nucleus);
+      LaTeXSettings.Placeholder.Color = System.Drawing.Color.Green;
+      Assert.Equal(RestingColorDefault, LaTeXSettings.Placeholder.Color);
+    }
+    [Fact]
+    public void DefaultPlaceholderAppearance() {
+      Assert.Null(LaTeXSettings.PlaceholderActiveColor);
+      Assert.Null(LaTeXSettings.PlaceholderRestingColor);
+      Assert.Equal(ActiveNucleusDefault, LaTeXSettings.PlaceholderActiveNucleus);
+      Assert.Equal(RestingNucleusDefault, LaTeXSettings.PlaceholderRestingNucleus);
+      Assert.Equal(LaTeXSettings.PlaceholderRestingNucleus, LaTeXSettings.Placeholder.Nucleus);
+      Assert.Equal(LaTeXSettings.PlaceholderRestingColor, LaTeXSettings.Placeholder.Color);
+    }
```

Comment:
Should split LaTeXSettings_Placeholder_IsNewInstance and DefaultPlaceholderAppearance into their own class under the same collection, and use the constructor and Dispose() for common init/deinit of the rest of tests.

Reply:
Also using underscores results in an inconsistent naming style.